File: identhub/src/main/java/de/solarisbank/identhub/data/verification/phone/VerificationPhoneNetworkDataSource.java
package de.solarisbank.identhub.data.verification.phone;

import de.solarisbank.identhub.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.data.verification.phone.model.VerificationPhoneResponse;
import io.reactivex.Single;

public interface VerificationPhoneNetworkDataSource {
    Single<VerificationPhoneResponse> postAuthorize();

    Single<VerificationPhoneResponse> postConfirm(final TransactionAuthenticationNumber tan);
}

File: phone/src/main/java/de/solarisbank/identhub/phone/data/VerificationPhoneNetworkDataSource.kt
package de.solarisbank.identhub.phone.data

import de.solarisbank.identhub.phone.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.phone.model.VerificationPhoneResponse
import io.reactivex.Single

interface VerificationPhoneNetworkDataSource {
    fun postAuthorize(): Single<VerificationPhoneResponse>
    fun postConfirm(tan: TransactionAuthenticationNumber): Single<VerificationPhoneResponse>
}